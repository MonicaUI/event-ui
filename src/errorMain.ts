import '../resources/less/error.less';
