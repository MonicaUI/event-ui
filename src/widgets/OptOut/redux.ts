/* eslint-disable import/no-cycle */

import { injectPageSnapshot } from '../../redux/actions';
import querystring from 'querystring';

/**
 * Opt-out and opt-in Redux module:
 * Event invitees can click "Opt Out" link in their emails to opt them out all communications from the event account
 */
const GET_OPT_OUT_STATUS = 'event-guestside-site/optOut/GET_OPT_OUT_STATUS';
const OPT_OUT_CONTACT = 'event-guestside-site/optOut/OPT_OUT_CONTACT';

const createGetOptedOutStatusAction = contactInfo => {
  return {
    type: GET_OPT_OUT_STATUS,
    payload: contactInfo
  };
};

export const getOptOutStatus =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    const optedOutContact = await eventGuestClient.getOptOutStatus(accessToken, eventId, inviteeId, guestId, contactId);
    dispatch(createGetOptedOutStatusAction(optedOutContact));
  };

export const createOptedOutAction = (contactInfo: $TSFixMe): $TSFixMe => {
  return {
    type: OPT_OUT_CONTACT,
    payload: contactInfo
  };
};

export const optOutAction =
  (optOutStatus: $TSFixMe, fromConfirmPage: $TSFixMe): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    // eslint-disable-next-line import/no-cycle
    const page = (await import(/* webpackChunkName optOut */ './content')).optInOutPageNew;
    if ((fromConfirmPage && optOutStatus) || !fromConfirmPage) {
      const optedOutContact = await eventGuestClient.optOut(
        accessToken,
        eventId,
        inviteeId,
        guestId,
        contactId,
        optOutStatus
      );
      dispatch(createOptedOutAction(optedOutContact));
    }
    dispatch(injectPageSnapshot(page));
  };

export const prepareForOptOutPageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    let page;
    if (!inviteeId && !guestId && !contactId) {
      page = (await import(/* webpackChunkName optOut */ './invalidInviteeContent')).invalidInviteeOptOut;
    } else {
      await dispatch(getOptOutStatus());
      if (getState() && getState().optOut && getState().optOut.optOutStatus) {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).optInOutPage;
      } else {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).confirmationPage;
      }
    }
    dispatch(injectPageSnapshot(page));
  };

export const optOutReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case GET_OPT_OUT_STATUS:
    case OPT_OUT_CONTACT:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
