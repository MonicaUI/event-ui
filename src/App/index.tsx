import React from 'react';
import { connect, useSelector } from 'react-redux';
import AppStyles from './App.less';
import AppContainer from 'nucleus-guestside-site/src/containers/AppContainer';
import PageRenderer from './PageRenderer';
import { getContentImages } from 'nucleus-themes/getImages';
import CheckoutProcessingDialog from '../dialogs/CheckoutProcessingDialog';
import NavigationDialog from './NavigationDialog';
import baseDialogStyle from '../dialogs/styles/Dialog.less';
import checkoutDialogStyles from '../dialogs/CheckoutProcessingDialog/CheckoutProcessingDialog.less';
import dialogTransitionUpStyle from 'event-widgets/shared-styles/DialogTransitionUp.less';
import Form from 'nucleus-form/src/components/Form';
import ThemeImagesContext from 'nucleus-widgets/context/ThemeImagesContext';
import DialogStylesContext from '../dialogs/DialogStylesContext';
import {
  CANCELLATION,
  DECLINE,
  GUEST_REGISTRATION,
  POST_REGISTRATION_PAYMENT,
  REGISTRATION,
  WAITLIST
} from '../redux/website/registrationProcesses';
import { useGraphQLSiteEditorData, GraphQLSiteEditorDataReleases } from '../ExperimentHelper';
import { RootState } from '../redux/reducer';
import { createSelector } from 'reselect';
import { Page } from '@cvent/flex-event-shared/target/guestside';
import { isEmpty } from 'lodash';
import useGraphQLPageData from './useGraphQLPageData';

/** Props for the App component */
export type AppProps = {
  assetRoot?: string;
  pageId?: string;
  page?: Page;
  templatePage?: Page;
  registrationPathId?: string;
  theme?: unknown;
  isPageUnderForm?: boolean;
  imageLookup?: unknown;
  browserFeatures?: {
    supportsWebp?: boolean;
  };
};

type AppState = {
  assetRoot?: string;
  themeImages?: unknown;
  dialogStyles?: {
    base: unknown;
    checkoutProcessing: unknown;
    transitions: {
      up: unknown;
    };
  };
  isPasswordModalOpen?: boolean;
};

/** Main app component to provide theme and dialog contexts and render guestside pages */
export class App extends React.Component<AppProps, AppState> {
  static displayName = 'App';

  constructor(props: AppProps) {
    super(props);
    this.state = {
      themeImages: {
        ...getContentImages(this.props.assetRoot)
      },
      dialogStyles: {
        base: baseDialogStyle,
        checkoutProcessing: checkoutDialogStyles,
        transitions: {
          up: dialogTransitionUpStyle
        }
      }
    };
  }

  static getDerivedStateFromProps(props: AppProps, state: AppState): AppState | null {
    if (props.assetRoot !== state.assetRoot) {
      return {
        assetRoot: props.assetRoot,
        themeImages: {
          ...getContentImages(props.assetRoot)
        }
      };
    }
    return null;
  }

  renderGuestPage(): JSX.Element {
    const { page, theme, isPageUnderForm, imageLookup, browserFeatures } = this.props;
    if (!page) {
      return null;
    }
    return (
      <AppContainer theme={theme} imageLookup={imageLookup} browserFeatures={browserFeatures}>
        {isPageUnderForm ? (
          <Form>
            <PageRenderer page={page} isPasswordModalOpen={this.state.isPasswordModalOpen} />
          </Form>
        ) : (
          <PageRenderer page={page} isPasswordModalOpen={this.state.isPasswordModalOpen} />
        )}
      </AppContainer>
    );
  }

  render(): JSX.Element {
    return (
      <ThemeImagesContext.Provider value={this.state.themeImages}>
        <DialogStylesContext.Provider value={this.state.dialogStyles}>
          <div className={AppStyles.wrapper}>
            {this.renderGuestPage()}
            <CheckoutProcessingDialog />
            {/* @ts-expect-error ts-migrate(2741) FIXME: Property 'isOpen' is missing in type '{}' but requ... Remove this comment to see the full error message */}
            <NavigationDialog />
          </div>
        </DialogStylesContext.Provider>
      </ThemeImagesContext.Provider>
    );
  }
}

/**
 * Create main application component, including page renderer, with a specific asset root URL
 */
export const createApp = (assetRoot: string): React.FunctionComponent<AppProps> => {
  return props => <AppExperimentWrapper {...props} assetRoot={assetRoot} />;
};

const EMPTY_OBJECT = Object.freeze({});
const selectImageLookup = createSelector(
  state => (state as RootState).imageLookup,
  imageLookup => {
    return isEmpty(imageLookup) ? EMPTY_OBJECT : imageLookup;
  }
);

const ConnectedApp = connect((state: RootState, props: AppProps) => {
  const { pageId } = props;
  return {
    theme: state.website.theme,
    isPasswordModalOpen: state.isPasswordModalOpen,
    isPageUnderForm:
      REGISTRATION.isTypeOfPage(state, pageId) ||
      GUEST_REGISTRATION.isTypeOfPage(state, pageId) ||
      CANCELLATION.isTypeOfPage(state, pageId) ||
      DECLINE.isTypeOfPage(state, pageId) ||
      WAITLIST.isTypeOfPage(state, pageId) ||
      POST_REGISTRATION_PAYMENT.isTypeOfPage(state, pageId),
    imageLookup: selectImageLookup(state),
    browserFeatures: state.browserFeatures
  };
})(App);

/** Wrapper component to pass down the current page from the Redux store */
export const AppWithRedux: React.FunctionComponent<AppProps> = (props: AppProps) => {
  const { pageId } = props;
  const pages = useSelector(state => (state as RootState).website.pages);
  const page = pages ? pages[pageId] : null;
  return <ConnectedApp {...props} page={page} />;
};

/** Wrapper component to pass down the current page from a GraphQL query */
export const AppWithGraphQL: React.FunctionComponent<AppProps> = (props: AppProps) => {
  const { pageId } = props;
  const page = useGraphQLPageData(pageId);
  return <ConnectedApp {...props} page={page} pageId={page?.id} />;
};

const AppExperimentWrapper: React.FunctionComponent<AppProps> = (props: AppProps) => {
  const shouldUseGraphQLSiteEditorData = useGraphQLSiteEditorData(GraphQLSiteEditorDataReleases.SiteEditorPageData);
  if (shouldUseGraphQLSiteEditorData) return <AppWithGraphQL {...props} />;
  return <AppWithRedux {...props} />;
};
