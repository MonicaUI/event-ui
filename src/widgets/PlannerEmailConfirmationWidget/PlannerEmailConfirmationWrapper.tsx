import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BaseWidget from 'nucleus-widgets/lib/BaseWidget';
import { REGISTRATION } from '../../redux/website/registrationProcesses';
import { getCurrentPageId } from '../../redux/pathInfo';

/**
 * A wrapper widget that wraps the Planner Email Confirmation and the Navigator widgets
 */
class PlannerEmailConfirmationWrapper extends BaseWidget<$TSFixMe, $TSFixMe> {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    style: PropTypes.any.isRequired,
    classes: PropTypes.any,
    isPlanner: (PropTypes as $TSFixMe).boolean,
    isLastPage: (PropTypes as $TSFixMe).boolean,
    NavigatorWidget: PropTypes.node
  };

  props: $TSFixMe;

  render() {
    const { isPlanner, isLastPage, NavigatorWidget } = this.props;
    const PlannerEmailConfirmationWidget = React.lazy(() => import('./PlannerEmailConfirmationWidget'));
    return (
      <div>
        {isLastPage && isPlanner && (
          <div>
            <Suspense fallback={<div></div>}>
              <PlannerEmailConfirmationWidget {...this.props} />
            </Suspense>
          </div>
        )}
        <NavigatorWidget {...this.props} />
      </div>
    );
  }
}

export default connect((state: $TSFixMe, props: $TSFixMe) => {
  const pageIds = REGISTRATION.forPathContainingWidget(props.id).pageIds(state);
  const currentPageId = getCurrentPageId(state);
  return {
    isPlanner: state.defaultUserSession.isPlanner,
    isLastPage:
      currentPageId.includes('registrationCancellationPage') ||
      currentPageId.includes('registrationDeclinePage') ||
      pageIds.indexOf(currentPageId) === pageIds.length - 1
  };
})(PlannerEmailConfirmationWrapper);
