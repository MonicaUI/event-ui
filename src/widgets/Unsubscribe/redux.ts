import { injectPageSnapshot } from '../../redux/actions';
import querystring from 'querystring';

/**
 * Unsubscribe and re-subscribe Redux module:
 * Event invitees can click "Unsubscribe" link in their emails to opt them out from the event emails communication
 */
const GET_UNSUBSCRIBE_STATUS = 'event-guestside-site/unsubscribe/GET_UNSUBSCRIBE_STATUS';
const SET_UNSUBSCRIBE_STATUS = 'event-guestside-site/unsubscribe/SET_UNSUBSCRIBE_STATUS';

const createGetUnsubscribeStatusAction = (unsubscribedContact, optOutStatus) => {
  return {
    type: GET_UNSUBSCRIBE_STATUS,
    payload: {
      ...optOutStatus,
      unsubscribeStatus: !unsubscribedContact.subscribed
    }
  };
};

export const getUnsubscribeStatus =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventEmailClient, eventGuestClient },
      accessToken,
      event: { id: eventId },
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const encodedGuestId = queryParams.g;
    const unsubscribedContact = await eventEmailClient.getSubscriptionStatus(eventId, inviteeId, encodedGuestId);
    // optOutStatus object has invitee/guest and planner infomation for display
    const optOutStatus = await eventGuestClient.getOptOutStatus(accessToken, eventId, inviteeId, encodedGuestId);
    dispatch(createGetUnsubscribeStatusAction(unsubscribedContact, optOutStatus));
  };

const createUnsubscribeAction = unsubscribeStatus => {
  return {
    type: SET_UNSUBSCRIBE_STATUS,
    payload: { unsubscribeStatus }
  };
};

export const unsubscribeAction =
  (unsubscribeStatus: $TSFixMe): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventEmailClient },
      event: { id: eventId },
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const encodedGuestId = queryParams.g;
    const subscribed = !unsubscribeStatus;
    await eventEmailClient.setSubscriptionStatus(eventId, inviteeId, encodedGuestId, subscribed);
    dispatch(createUnsubscribeAction(unsubscribeStatus));
  };

export const prepareForUnsubscribePageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const encodedGuestId = queryParams.g;
    let unsubscribePage;
    if (!inviteeId && !encodedGuestId) {
      unsubscribePage = (await import(/* webpackChunkName: "unsubscribe" */ './invalidInviteeContent'))
        .invalidInviteeUnsubscribePage;
    } else {
      await dispatch(getUnsubscribeStatus());
      // eslint-disable-next-line import/no-cycle
      unsubscribePage = (await import(/* webpackChunkName: "unsubscribe" */ './content')).unsubscribePage;
    }
    dispatch(injectPageSnapshot(unsubscribePage));
  };

export const unsubscribeReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case GET_UNSUBSCRIBE_STATUS:
    case SET_UNSUBSCRIBE_STATUS:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
