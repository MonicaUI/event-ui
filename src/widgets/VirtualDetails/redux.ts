import querystring from 'querystring';
import { injectPageSnapshot } from '../../redux/actions';
import { populateAllProducts } from '../../redux/visibleProducts';
import { isBeforeStartTime, currentTimeIsInRangeOf } from 'event-widgets/redux/modules/timezones';
import { CANCELLED } from 'event-widgets/clients/EventStatus';
import { sessionStatus } from 'event-widgets/redux/selectors/event';
import Logger from '@cvent/nucleus-logging';

const SET_VIRTUAL_DETAILS = 'event-guestside-site/virtualDetails/SET_VIRTUAL_DETAILS';
const DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE =
  'event-guestside-site/virtualDetails/DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE';
export const EVENT = 'event';
export const SESSION = 'session';
export const VIRTUAL_DETAILS_PAGE = 'virtualDetailsPage';
export const SESSION_NOT_AVAILABLE_PAGE = 'sessionNotAvailableMessagePage';
const LOG = new Logger('widgets/virtualDetails/redux');

const setVirtualDetails = ({ code, url, type, isPassed, pageToDisplay }) => {
  return {
    type: SET_VIRTUAL_DETAILS,
    payload: { code, url, type, isPassed, pageToDisplay }
  };
};

const displaySessionNotAvailableMessage = () => {
  return {
    type: DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE,
    payload: { pageToDisplay: SESSION_NOT_AVAILABLE_PAGE }
  };
};

export const joinWebcastSession =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      virtualDetails: { url }
    } = getState();
    if (url) {
      window.open(url, '_blank');
    }
  };

const getURLAndCode = (attendeeWebcastData, genericWebcastData) => {
  let url;
  let code;

  if (attendeeWebcastData?.join?.href) {
    url = attendeeWebcastData?.join?.href;
    code = attendeeWebcastData?.join?.code;
  } else if (genericWebcastData?.joiningURL) {
    url = genericWebcastData?.joiningURL;
    code = genericWebcastData?.joiningCode;
  }

  return { url, code };
};

const fetchSessionWebcastData = (attendeeId, sessionId) => async (dispatch, getState) => {
  let {
    // eslint-disable-next-line prefer-const
    selectedTimeZone,
    // eslint-disable-next-line prefer-const
    text: { eventTimezone },
    // eslint-disable-next-line prefer-const
    clients: { universalWebcastClient },
    visibleProducts: { Widget: { sessionProducts = {} } = {} }
  } = getState();
  let virtualData = {};
  let attendeeLinksContainer;
  let attendeeWebcastData;

  await dispatch(populateAllProducts('Widget'));
  sessionProducts = getState().visibleProducts?.Widget?.sessionProducts;
  const selectedSessionDetails = sessionProducts[sessionId];

  // If session was deleted in test mode
  if (!selectedSessionDetails) {
    dispatch(displaySessionNotAvailableMessage());
    return;
  }

  if (selectedSessionDetails.status === sessionStatus.cancelled) {
    return;
  }

  const isSessionPassed = !(
    currentTimeIsInRangeOf(
      selectedSessionDetails.startTime,
      selectedSessionDetails.endTime,
      selectedTimeZone,
      eventTimezone
    ) || isBeforeStartTime(selectedSessionDetails.startTime, selectedTimeZone, eventTimezone)
  );

  if (isSessionPassed) {
    if (selectedSessionDetails?.webcast?.recordingURL) {
      virtualData = {
        url: selectedSessionDetails?.webcast?.recordingURL,
        code: selectedSessionDetails?.webcast.recordingCode
      };
    } else {
      return;
    }
  } else {
    try {
      attendeeLinksContainer = await universalWebcastClient.getAttendeeLinkData(attendeeId);
    } catch (error) {
      LOG.warn(`Failed to get attendee specific session webcast data for attendeeId ${attendeeId}`);
      return;
    }
    attendeeWebcastData = attendeeLinksContainer?.data.find(
      attendeeLinkData => attendeeLinkData?.session?.id === sessionId
    );
    virtualData = getURLAndCode(attendeeWebcastData, selectedSessionDetails.webcast);
  }

  if (!(virtualData as $TSFixMe)?.url) {
    return;
  }

  dispatch(
    // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ type: string; isPassed: boolea... Remove this comment to see the full error message
    setVirtualDetails({ ...virtualData, type: SESSION, isPassed: isSessionPassed, pageToDisplay: VIRTUAL_DETAILS_PAGE })
  );
};

const fetchEventWebcastData = attendeeId => async (dispatch, getState) => {
  const {
    event: { status: eventStatus, webcast: eventWebcast, startDate, endDate },
    selectedTimeZone,
    text: { eventTimezone },
    clients: { universalWebcastClient }
  } = getState();
  let virtualData = {};
  let attendeeLinksContainer;
  let attendeeWebcastData;

  if (eventStatus === CANCELLED) {
    return;
  }

  const isEventPassed = !(
    currentTimeIsInRangeOf(startDate, endDate, selectedTimeZone, eventTimezone) ||
    isBeforeStartTime(startDate, selectedTimeZone, eventTimezone)
  );

  if (isEventPassed) {
    if (eventWebcast?.recordingURL) {
      virtualData = {
        url: eventWebcast?.recordingURL,
        code: eventWebcast?.recordingCode
      };
    } else {
      return;
    }
  } else {
    try {
      attendeeLinksContainer = attendeeId ? await universalWebcastClient.getAttendeeLinkData(attendeeId) : undefined;
    } catch (error) {
      LOG.warn(`Failed to get attendee specific event webcast data for attendeeId ${attendeeId}`);
      return;
    }
    attendeeWebcastData = attendeeLinksContainer?.data.find(
      attendeeLinkData => Object.keys(attendeeLinkData?.session).length === 0
    );
    virtualData = getURLAndCode(attendeeWebcastData, eventWebcast);
  }

  if (!(virtualData as $TSFixMe)?.url) {
    return;
  }

  dispatch(
    // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ type: string; isPassed: boolea... Remove this comment to see the full error message
    setVirtualDetails({ ...virtualData, type: EVENT, isPassed: isEventPassed, pageToDisplay: VIRTUAL_DETAILS_PAGE })
  );
};

export const prepareForVirtualDetailsPageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const sessionId = queryParams.sessionId;
    const attendeeId = guestId || inviteeId;
    let virtualDetailsPage;

    if (sessionId) {
      if (attendeeId) {
        await dispatch(fetchSessionWebcastData(attendeeId, sessionId));
      }
    } else {
      await dispatch(fetchEventWebcastData(attendeeId));
    }

    if (getState()?.virtualDetails?.pageToDisplay === VIRTUAL_DETAILS_PAGE) {
      // eslint-disable-next-line import/no-cycle
      virtualDetailsPage = (await import(/* webpackChunkName: "virtualDetails" */ './content')).virtualDetailsPage;
    } else if (getState()?.virtualDetails?.pageToDisplay === SESSION_NOT_AVAILABLE_PAGE) {
      // eslint-disable-next-line import/no-cycle
      virtualDetailsPage = (await import(/* webpackChunkName: "virtualDetails" */ './content'))
        .sessionNotAvailableMessagePage;
    } else {
      // eslint-disable-next-line import/no-cycle
      virtualDetailsPage = (await import(/* webpackChunkName: "virtualDetails" */ './content')).genericMessagePage;
    }

    dispatch(injectPageSnapshot(virtualDetailsPage));
  };

export const virtualDetailsReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case SET_VIRTUAL_DETAILS:
    case DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
