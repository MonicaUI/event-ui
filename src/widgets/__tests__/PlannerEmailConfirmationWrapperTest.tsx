import React from 'react';
import renderer from 'react-test-renderer';
import PlannerEmailConfirmationWrapper from '../PlannerEmailConfirmationWidget/PlannerEmailConfirmationWrapper';
import RegistrationNavigatorWidget from '../RegistrationNavigator/RegistrationNavigatorWidget';
import { createStore } from 'redux';
import { createLocalNucleusForm } from 'nucleus-form/src/NucleusForm';

jest.mock('../../redux/website/registrationProcesses', () => {
  return {
    REGISTRATION: {
      forPathContainingWidget: () => {
        return {
          pageIds: () => {
            return ['67890', '12345'];
          }
        };
      }
    }
  };
});

jest.mock('../../ExperimentHelper', () => ({
  ...jest.requireActual<$TSFixMe>('../../ExperimentHelper'),
  useGraphQLForSkippingPages: () => false
}));

const defaultProps = {
  nucleusForm: createLocalNucleusForm(),
  classes: {},
  style: {},
  translate: jest.fn(x => x),
  config: {
    displayText: {
      backward: 'backward',
      forward: 'forward',
      complete: 'complete',
      exit: 'exit'
    }
  }
};

let initialState = {};

describe('PlannerEmailConfirmationWrapper', () => {
  beforeEach(() => {
    initialState = {
      text: {
        translate: () => jest.fn(x => x)
      },
      defaultUserSession: {
        isPlanner: true
      },
      pathInfo: {
        currentPageId: '12345'
      },
      website: {
        theme: {
          global: {
            elements: {}
          }
        }
      },
      registrationForm: {
        regCart: {
          sendEmail: true
        }
      }
    };
    jest.clearAllMocks();
  });
  test('should render with no checkbox visible on non-last page', () => {
    (initialState as $TSFixMe).pathInfo.currentPageId = '67890';
    const widget = renderer.create(
      <PlannerEmailConfirmationWrapper
        {...defaultProps}
        store={createStore((state = initialState) => {
          return state;
        })}
        NavigatorWidget={RegistrationNavigatorWidget}
      />
    );
    expect(widget).toMatchSnapshot();
  });
  test('should render with no checkbox visible when not planner', () => {
    (initialState as $TSFixMe).defaultUserSession.isPlanner = false;
    const widget = renderer.create(
      <PlannerEmailConfirmationWrapper
        {...defaultProps}
        store={createStore((state = initialState) => {
          return state;
        })}
        NavigatorWidget={RegistrationNavigatorWidget}
      />
    );
    expect(widget).toMatchSnapshot();
  });
  test('pre-load checkbox widget for test', () => {
    const widget = renderer.create(
      <PlannerEmailConfirmationWrapper
        {...defaultProps}
        store={createStore((state = initialState) => {
          return state;
        })}
        NavigatorWidget={RegistrationNavigatorWidget}
      />
    );
    expect(widget).toMatchSnapshot();
  });
  test('should render with checkbox visible when planner', () => {
    const widget = renderer.create(
      <PlannerEmailConfirmationWrapper
        {...defaultProps}
        store={createStore((state = initialState) => {
          return state;
        })}
        NavigatorWidget={RegistrationNavigatorWidget}
      />
    );
    expect(widget).toMatchSnapshot();
  });
});
