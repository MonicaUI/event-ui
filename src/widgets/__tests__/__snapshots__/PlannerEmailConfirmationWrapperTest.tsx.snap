// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlannerEmailConfirmationWrapper pre-load checkbox widget for test 1`] = `
<div>
  <div>
    <div />
  </div>
  <div
    className=""
    style={
      Object {
        "flex": "1 1 auto",
        "pointerEvents": undefined,
      }
    }
  >
    <ul
      className=""
      style={Object {}}
    >
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="backward"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          backward
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="exit"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          exit
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="complete"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          complete
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PlannerEmailConfirmationWrapper should render with checkbox visible when planner 1`] = `
<div>
  <div>
    <div />
  </div>
  <div
    className=""
    style={
      Object {
        "flex": "1 1 auto",
        "pointerEvents": undefined,
      }
    }
  >
    <ul
      className=""
      style={Object {}}
    >
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="backward"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          backward
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="exit"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          exit
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="complete"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          complete
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PlannerEmailConfirmationWrapper should render with no checkbox visible on non-last page 1`] = `
<div>
  <div
    className=""
    style={
      Object {
        "flex": "1 1 auto",
        "pointerEvents": undefined,
      }
    }
  >
    <ul
      className=""
      style={Object {}}
    >
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="exit"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          exit
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="forward"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          forward
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PlannerEmailConfirmationWrapper should render with no checkbox visible when not planner 1`] = `
<div>
  <div
    className=""
    style={
      Object {
        "flex": "1 1 auto",
        "pointerEvents": undefined,
      }
    }
  >
    <ul
      className=""
      style={Object {}}
    >
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="backward"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          backward
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="exit"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          exit
        </button>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <button
          className=""
          disabled={false}
          id="complete"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          complete
        </button>
      </li>
    </ul>
  </div>
</div>
`;
